# PIPE-3431-HiC-Data-Analysis

Scripts for running CHiCAGO for promoter capture Hi-C data analysis. 

## Internal SIH

* Client: Marina Kennerson, Ally Boyling (PhD student)
* Support: consult
* [JIRA ticket](https://ctdshub.atlassian.net/browse/PIPE-3431?atlOrigin=eyJpIjoiNjQ3NjI4MDFiMmNkNDRkZWI3MDYyZDkyYTc2NTAyYjIiLCJwIjoiaiJ9)

## Progress notes 

### Meeting 14/10/22

* Code management with GitHub
* Workflow overview considerations
  * Run the HiCUP container at Artemis (QC & alignment)  
  * v0.8.2 installed on Artemis 
  * Try with test dataset from 
* Tool installation 
* Trial run with [test data]() 

## Resources

* [HiCUP github](https://github.com/StevenWingett/HiCUP)
* [HiCUP guide](https://www.bioinformatics.babraham.ac.uk/projects/hicup/read_the_docs/html/index.html)
* [Capture scripts](https://dovetail-capture.readthedocs.io/en/latest/index.html)
* [CHIiCAGO paper](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0992-2)
* [CHiCAGO protocols paper](https://www.nature.com/articles/s41596-021-00567-5) 
* [CHiCAGO bitbucket](https://bitbucket.org/chicagoTeam/chicago/src/master/)
* [chicdiff guide](https://dovetail-capture.readthedocs.io/en/latest/Interaction_analysis.html#diff)
* [chicdiff github](https://github.com/RegulatoryGenomicsGroup/chicdiff/)
